### ISSUE TITLE
Autoroute Pattern and {{ | t }} filter

### ISSUE THREAD

**Issue by Skrypt** [OrchardCMS/OrchardCore#5599]
We cannot use {{ "something" | t }} in an autoroute pattern because it will always use the CultureInfo.CurrentUICulture. I tried forcing a Culture on the LiquidTemplateContext and it doesn't take it into consideration.

**Comment by Skrypt** [comment]
![image](https://user-images.githubusercontent.com/3228637/74957993-4e4d0a80-53d6-11ea-990a-5c7351a3e3c9.png)

I did this ugly patch to fix the issue but we might want eventually to do this better.

**Comment by hishamco** [comment]
> Probably because initially the PortableObjectStringLocalizer doesn't take any culture in param

This is weird, we could use `WithCulture(CultureInfo culture)` until 4.0 is shipped, but I need to understand more about this issue, why you need to enforce the culture?

It is because you need to get the proper content translation

**Comment by Skrypt** [comment]
The translated string is not taken from the content. I'm using a global translated string in my autoroute pattern and I want it to be translated in the culture of the content item. So, here I'm using a LocalizationPart on a content item and I want that my autoroute be saved with some french or english patterns. Example : /category/apple/{{ contentitemId }} --> /categorie/pomme/{{ contentitemId }}

**Comment by hishamco** [comment]
In nutshell what you want did is "Slug/ Route Localization"?

**Comment by Skrypt** [comment]
Yes AutoroutePart pattern localization. Right now we can't because it's always using the CultureInfo.CurrentUICulture when we parse a Liquid template. It should use the LocalizationPart.Culture to properly translate in the correct culture.

**Comment by hishamco** [comment]
Ya,  could use `WithCulture(CultureInfo culture)` for now until 4.0 is shipped, also I will try to think about this one

**Comment by hishamco** [comment]
If you want me to try this one, I can start a PR

**Comment by Skrypt** [comment]
Go for it have fun 😄

**Comment by sebastienros** [comment]
`ICultureAspect`

**Comment by sebastienros** [comment]
You will need to call Render with a custom instance of `LiquidTemplateContext` (new LiquidTemplateContext(ShellScope.Services)) and then pass it the culture of the content item.

**Comment by hishamco** [comment]
I really forgot `ICultureAspect` ;)

**Comment by Skrypt** [comment]
```
                var customLiquidTemplateContext = new LiquidTemplateContext(ShellScope.Services)
                {
                    CultureInfo = CultureInfo.CurrentUICulture
                };

                if (part.ContentItem.Content.LocalizationPart.Culture != null)
                {
                    customLiquidTemplateContext.CultureInfo = CultureInfo.GetCultureInfo(part.ContentItem.Content.LocalizationPart.Culture.Value);
                }

                part.Path = await _liquidTemplateManager.RenderAsync(pattern, customLiquidTemplateContext, NullEncoder.Default, model,
                    scope => scope.SetValue("ContentItem", model.ContentItem));
```

```
        public Task<string> RenderAsync(string source, LiquidTemplateContext context, TextEncoder encoder, object model, Action<Scope> scopeAction)
        {
            if (String.IsNullOrWhiteSpace(source))
            {
                return Task.FromResult((string)null);
            }

            var result = GetCachedTemplate(source);

            return result.RenderAsync(encoder, context, model, scopeAction);
        }
```

Same result.

**Comment by Skrypt** [comment]
Ended up doing this for now which works : 

```
                var currentCulture = CultureInfo.CurrentUICulture;
                if (part.ContentItem.As<LocalizationPart>() != null)
                {
                    CultureInfo.CurrentUICulture = CultureInfo.GetCultureInfo(part.ContentItem.As<LocalizationPart>().Culture);
                }

                part.Path = await _liquidTemplateManager.RenderAsync(pattern, NullEncoder.Default, model,
                    scope => scope.SetValue("ContentItem", model.ContentItem));

                if (part.ContentItem.As<LocalizationPart>() != null)
                {
                    CultureInfo.CurrentUICulture = currentCulture;
                }
```

**Comment by hishamco** [comment]
@Skrypt seems you create a custom `RenderAsync()` version, which I don't have in the latest bits, is that for a reason? Is the intend to use it in `UpdatedAsync()`?

**Comment by Skrypt** [comment]
No in the latest bits I don't do anything else than the changes I made in my last post. I tried doing an instanciation of the LiquidTemplateContext like @sebastienros suggested but it doesn't take in consideration the LiquidTemplateContext.Culture at all. So it is using the CultureInfo.CurrentUICulture somewhere else down the line. I've found a place where we explicitly set it that way but not sure yet.
